### PR TITLE
Added an integration test, testing the whole composer

### PR DESCRIPTION
**Pull request by AndreasKl**

### CODE REVIEW

File: src/main/java/com/rewedigital/composer/ComposerApplication.java [hunk 1]
```diff
@@ -20,33 +20,47 @@
...
+    static class Initializer {
 
-        private static void init(final Environment environment) {
+        static void init(final Environment environment) {
             final Config configuration = withDefaults(environment.config());
 
             final ComposingRequestHandler handler =
                 new ComposingRequestHandler(
-                    new BackendRouting(configuration.getConfig("composer.routing")),
+                    new BackendRouting(configuration.getConfig(COMPOSER + ".routing")),
```
Comment (thovid) [inline src/main/java/com/rewedigital/composer/ComposerApplication.java]: I'm not sure if the constant `COMPOSER` improves readability...

File: src/main/java/com/rewedigital/composer/configuration/DefaultConfiguration.java [hunk 2]
```diff
@@ -10,14 +10,14 @@
 
 public class DefaultConfiguration {
 
-    private static final Map<String, Object> defaultConfiguration = initDefaultConfigurationMap();
+    private static final Map<String, Object> DEFAULT_CONFIGURATION = initDefaultConfigurationMap();
```
Comment (thovid) [inline src/main/java/com/rewedigital/composer/configuration/DefaultConfiguration.java]: During the last years, I have grown the feeling that the ALL_CAPS convention for `private static final` fields sometimes decreases readability. For example, In this particular case I don't think that the constant aspect of the field is important. It's just an optimization and should not be overemphasized. Thus I liked the normal camelCase naming more

File: src/test/java/com/rewedigital/composer/CompositionTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+import okio.ByteString;
+
+public class CompositionTest {
+
+    @Rule
+    public final MockWebServer server = new MockWebServer();
+
+    @Rule
+    public ServiceHelper serviceHelper =
```
Comment (thovid) [inline src/test/java/com/rewedigital/composer/CompositionTest.java:28]: I really like this way to "start" the service!

File: src/test/java/com/rewedigital/composer/CompositionTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        ServiceHelper.create(ComposerApplication.Initializer::init, ComposerApplication.COMPOSER)
+            .conf("composer.routing.local-routes", routesConfig(mockServerUrl()))
```
Comment (thovid) [inline src/test/java/com/rewedigital/composer/CompositionTest.java]: After you have introduced the constant `ComposerApplication.COMPOSER`, shouldn't you use it here? But on the other hand, `ComposerApplication.COMPOSER + "routing.local-routes"` is so much worse regarding readability, that I don't like the constant at all ;-)

File: src/test/java/com/rewedigital/composer/CompositionTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            .conf("composer.html.include-tag", "include")
+            .conf("composer.html.content-tag", "content");
+
+    @Test
+    public void parsesMasterTemplateAndCombinesChildTemplate() throws Exception {
```
Comment (thovid) [inline src/test/java/com/rewedigital/composer/CompositionTest.java]: When I read a test I like to see the connection between inputs and expected output. Here, the expected value `"4223"` appears out of nowhere... So, maybe an assertion that raises the level of abstraction by using a `Condition` is more appropriate: `asserThat(responseBody(composedFuture)).is(theCompositionOfMockServiceResponses())` or something like that?

**Review by thovid — APPROVED** [review]
I really like the `@Rule` to setup the service instance. 

My biggest issue is with the ALL_CAPS name of the `private static final` field as I think it reduces a readability by overemphasizing the wrong part... Here we should come to an agreement regarding the coding style.

I also don't think extracting the `ComposerApplication.COMPOSER` constant makes sense. If we stick to it, we should use it everywhere which would create a lot of dependencies on `ComposerApplication` throughout the code...

File: src/main/java/com/rewedigital/composer/ComposerApplication.java (hunk 1 above)
Comment (AndreasKl) [reply src/main/java/com/rewedigital/composer/ComposerApplication.java]: Agree..

File: src/main/java/com/rewedigital/composer/configuration/DefaultConfiguration.java (hunk 2 above)
Comment (AndreasKl) [reply src/main/java/com/rewedigital/composer/configuration/DefaultConfiguration.java]: Let's talk in person on Monday about that. Reverted the change.

**Pull request merged**